src/lib/rnp2.cpp

```diff
@@ -3152,15 +3155,16 @@ key_get_uid_at(pgp_key_t *key, size_t idx, char **uid)
     if (!key || !uid) {
         return RNP_ERROR_NULL_POINTER;
     }
-    if (idx >= key->uidc || DYNARRAY_IS_EMPTY(key, uid)) {
+    if (idx >= pgp_get_userid_count(key)) {
```

Comment:
This removes the check for UIDs being empty, does it matter?

Reply:
@ni4 could you have a look here? Thanks!

Reply:
Empty uid list equals to `pgp_get_userid_count() == 0`, which will satisfy the `idx >=` check  since idx is unsigned.

Reply:
Agree with @ni4, this should be good. It looks like the `DYNARRAY_IS_EMPTY` check here before was bogus (added by yours truly :P).

Reply:
Thanks guys!